File: Sources/CountriesFlagsObjC/source/Countries/UIImage+Vietnam.m
//
//  UIImage+Vietnam.m
//  CountriesFlags
//

#import "Countries/UIImage+Vietnam.h"

#import "Extensions/UIBezierPath+Star.h"
#import "Extensions/CGSize+Extension.h"

@implementation UIImage (Vietnam)

+ (UIImage *)flagVietnamWithSize:(CGSize)size {
    UIColor* colorYellow = [UIColor yellowColor];
    UIColor* colorRed = [UIColor colorWithRed:218.0f/255.0f
                                        green:37.0f/255.0f
                                         blue:29.0f/255.0f
                                        alpha:1.0f];
    
    UIGraphicsBeginImageContextWithOptions(size, YES, 0.0f);
    
    [colorRed setFill];
    UIRectFill(CGRectMake(0.0f, 0.0f, size.width, size.height));
    
    [colorYellow setFill];
    [[UIBezierPath starAtPoint:CGSizeMidPoint(size)
                        radius:size.height * 12.0f / 40.0f
                          type:FlagStarTypeFivePointed] fill];
    
    UIImage* image = UIGraphicsGetImageFromCurrentImageContext();
    UIGraphicsEndImageContext();
    
    return image != nil ? image : [UIImage new];
}

@end

File: Sources/CountriesFlags/Countries/UIImage+Vietnam.swift
//
//  UIImage+Vietnam.swift
//  CountriesFlags
//

#if canImport(UIKit)

import UIKit

extension UIImage {
    /**
     [wikipedia]: https://en.wikipedia.org/wiki/Flag_of_Vietnam "wikipedia"
     
     Generates the flag of Vietnam.
     For more information see [wikipedia].

     - Parameters:
        - size: image size.
     
     - Returns: Returns the flag of Vietnam.
     */
    internal static func flagVietnam(with size: CGSize) -> UIImage {
        let colorYellow = UIColor.yellow
        let colorRed = UIColor(red: 218.0 / 255.0,
                               green: 37.0 / 255.0,
                               blue: 29.0 / 255.0,
                               alpha: 1.0)

        UIGraphicsBeginImageContextWithOptions(size, true, 0.0)

        colorRed.setFill()
        UIRectFill(CGRect(origin: .zero, size: size))

        colorYellow.setFill()
        UIBezierPath.star(at: size.midPoint, radius: size.height * 12.0 / 40.0).fill()

        let image = UIGraphicsGetImageFromCurrentImageContext()
        UIGraphicsEndImageContext()

        return image ?? UIImage()
    }
}

#endif
